satpy/writers/cf_writer.py

```diff
@@ -576,54 +711,341 @@ def update_encoding(dataset, to_netcdf_kwargs, numeric_name_prefix='CHANNEL_'):
+    """Add 'history' attribute to dictionary."""
     _history_create = 'Created by pytroll/satpy on {}'.format(datetime.utcnow())
-    if 'history' in root.attrs:
-        if isinstance(root.attrs['history'], list):
-            root.attrs['history'] = ''.join(root.attrs['history'])
-        root.attrs['history'] += '\n' + _history_create
+    if 'history' in attrs:
+        if isinstance(attrs['history'], list):
+            attrs['history'] = ''.join(attrs['history'])
+        attrs['history'] += '\n' + _history_create
     else:
-        root.attrs['history'] = _history_create
+        attrs['history'] = _history_create
+    return attrs
+
 
+def _get_groups(groups, list_datarrays):
+    """Return a dictionary with the list of xr.DataArray associated to each group.
 
-def _get_groups(groups, datasets, root):
+    If no groups (groups=None), return all DataArray attached to a single None key.
+    Else, collect the DataArrays associated to each group.
+    """
     if groups is None:
-        # Groups are not CF-1.7 compliant
-        if 'Conventions' not in root.attrs:
-            root.attrs['Conventions'] = CF_VERSION
-        # Write all datasets to the file root without creating a group
-        groups_ = {None: datasets}
+        grouped_dataarrays = {None: list_datarrays}
     else:
-        # User specified a group assignment using dataset names. Collect the corresponding datasets.
-        groups_ = defaultdict(list)
-        for dataset in datasets:
+        grouped_dataarrays = defaultdict(list)
+        for datarray in list_datarrays:
             for group_name, group_members in groups.items():
-                if dataset.attrs['name'] in group_members:
-                    groups_[group_name].append(dataset)
+                if datarray.attrs['name'] in group_members:
+                    grouped_dataarrays[group_name].append(datarray)
                     break
-    return groups_
+    return grouped_dataarrays
+
+
+def make_cf_dataarray(dataarray,
+                      epoch=EPOCH,
+                      flatten_attrs=False,
+                      exclude_attrs=None,
+                      include_orig_name=True,
+                      numeric_name_prefix='CHANNEL_'):
+    """Make the xr.DataArray CF-compliant.
+
+    Parameters
+    ----------
+    dataarray : xr.DataArray
+        The data array to be made CF-compliant.
+    epoch : str, optional
+        Reference time for encoding of time coordinates.
+    flatten_attrs : bool, optional
+        If True, flatten dict-type attributes.
+        The default is False.
+    exclude_attrs : list, optional
+        List of dataset attributes to be excluded.
+        The default is None.
+    include_orig_name : bool, optional
+        Include the original dataset name in the netcdf variable attributes.
+        The default is True.
+    numeric_name_prefix : TYPE, optional
+        Prepend dataset name with this if starting with a digit.
+        The default is 'CHANNEL_'.
```

Comment:
RTD docs generation is failing with:

```
/home/docs/checkouts/readthedocs.org/user_builds/satpy/conda/2259/lib/python3.10/site-packages/satpy/writers/cf_writer.py:docstring of satpy.writers.cf_writer.make_cf_dataarray:16: ERROR: Unknown target name: "channel".
```

I think it means the below even though that's not line 16 of the docstring.

```suggestion
        The default is "CHANNEL_".
```